src/PackageDetailsViewExtension/PackageDetailItem.cs

```diff
@@ -152,15 +167,18 @@ public bool CanInstall
         /// <param name="packageName"></param>
         /// <param name="packageVersion"></param>
         /// <param name="canInstall"></param>
-        public PackageDetailItem(string packageName, PackageVersion packageVersion, bool canInstall)
+        /// <param name="isEnabledForInstall">True, if package is not already downloaded, is not deprecated, and package loading is allowed.</param>
+        public PackageDetailItem(string packageName, PackageVersion packageVersion, bool canInstall, bool isEnabledForInstall = true)
         {
             this.PackageName = packageName;
             this.PackageVersion = packageVersion;
             this.PackageVersionNumber = PackageVersion.version;
             this.CopyRightHolder = PackageVersion.copyright_holder;
             this.CopyRightYear = PackageVersion.copyright_year;
             this.CanInstall = canInstall;
-            
+            this.IsEnabledForInstall = isEnabledForInstall && canInstall;
```

Comment:
should we use this statement `isEnabledForInstall && canInstall` in the getter instead so the value is always up to date?

Reply:
Do you mean like this:
```
public bool IsEnabledForInstall
{
    get => isEnabledForInstall && canInstall;
    set
    {
        isEnabledForInstall = value;
        RaisePropertyChanged(nameof(IsEnabledForInstall));
    }
}
```
and set like:
```
this.IsEnabledForInstall = isEnabledForInstall;
```

Reply:
On a second thought, if the setter and getter are pointing to different values, it is also confusing. I guess I am just confused that we have two booleans which are very similar, maybe name them slightly more obvious to each case?

Reply:
I actually used the same variable names to be consistent with the PackageManagerSearch window logic,
`canInstall` handles if the package is already loaded and `IsEnabledForInstall ` handles preference settings toggle value.